vendor/models/profile.py

```diff
@@ -206,4 +206,13 @@ def get_payment_sum(self):
         return self.payments.filter(deleted=False, status=PurchaseStatus.SETTLED).aggregate(Sum('amount'))
 
     def get_settled_payments(self):
-        return self.payments.filter(deleted=False, status=PurchaseStatus.SETTLED).order_by('amount', 'submitted_date')
\ No newline at end of file
+        return self.payments.filter(deleted=False, status=PurchaseStatus.SETTLED).order_by('amount', 'submitted_date')
+
+    def is_on_trial(self, offer):
+
+        on_trial_receipt = [receipt for receipt in self.get_active_offer_receipts(offer) if receipt.transaction and 'trial' in receipt.transaction]
```

Comment:
could we do something like `next((receipt for receipt in self.get_active_offer_receipts(offer) if receipt.transaction and 'trial' in receipt.transaction), None)`?

then it wouldnt have to loop through all the receipts necessarily, it would just go until it found one that fit the condition or default to None if it didnt find anything. not a big deal but might be worth checking out